src/main/java/net/dv8tion/jda/api/entities/TriggerType.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.entities;
+
+/**
+ * This is basically informs you what cased the auto-moderation system to be triggered.
+ */
+public enum TriggerType
+{
+    /**
+     * This checks if the content sent by a member contains a specific word.
+     * <br>
+     * The max per guild is 3.
+     */
+    KEYWORD(1, 3),
+    /**
+     * As the name suggests it checks for any links which are found to be harmful.
+     *
+     * <br>
+     * The max per guild is 1.
+     */
+    HARMFUL_LINK(2, 1),
```

Comment:
Should mention that the `HARMFUL_LINK` trigger type is not fully released [as described here](https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-trigger-types)